api/src/main/java/org/openmrs/module/commonreports/reports/ConditionsReportManager.java

```diff
@@ -0,0 +1,126 @@
+package org.openmrs.module.commonreports.reports;
+
+import java.io.InputStream;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import org.apache.commons.io.IOUtils;
+import org.openmrs.module.commonreports.ActivatedReportManager;
+import org.openmrs.module.initializer.api.InitializerService;
+import org.openmrs.module.reporting.common.DateUtil;
+import org.openmrs.module.reporting.common.MessageUtil;
+import org.openmrs.module.reporting.dataset.definition.SqlDataSetDefinition;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.openmrs.module.reporting.report.ReportDesign;
+import org.openmrs.module.reporting.report.definition.ReportDefinition;
+import org.openmrs.module.reporting.report.manager.ReportManagerUtil;
+import org.openmrs.util.OpenmrsClassLoader;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.openmrs.annotation.OpenmrsProfile;
+
+@Component
+@OpenmrsProfile(modules = { "emrapi:1.5 - 1.2*" })
+public class ConditionsReportManager extends ActivatedReportManager {
+	
+	@Autowired
+	private InitializerService inizService;
+	
+	@Override
+	public boolean isActivated() {
+		return inizService.getBooleanFromKey("report.conditions.active", false);
+	}
+	
+	@Override
+	public String getVersion() {
+		return "1.0.0-SNAPSHOT";
+	}
+	
+	@Override
+	public String getUuid() {
+		return "e193acbb-4f12-43be-97b2-25ed70a3face";
+	}
+	
+	@Override
+	public String getName() {
+		return MessageUtil.translate("commonreports.report.conditions.reportName");
+	}
+	
+	@Override
+	public String getDescription() {
+		return MessageUtil.translate("commonreports.report.conditions.reportDescription");
+	}
+	
+	private Parameter getStartDateParameter() {
+		return new Parameter("onsetDate", "Onset Date", Date.class, null, DateUtil.parseDate("1970-01-01", "yyyy-MM-dd"));
+	}
+	
+	private Parameter getEndDateParameter() {
+		String today = new SimpleDateFormat("yyyy-MM-dd").format(new Date());
+		return new Parameter("endDate", "End Date", Date.class, null, DateUtil.parseDate(today, "yyyy-MM-dd"));
+	}
+	
+	private String getSqlString(String resourceName) {
+		
+		InputStream is = null;
+		try {
+			is = OpenmrsClassLoader.getInstance().getResourceAsStream(resourceName);
+			return IOUtils.toString(is, "UTF-8");
+		}
+		catch (Exception e) {
+			throw new IllegalArgumentException("Unable to load resource: " + resourceName, e);
+		}
+		finally {
+			IOUtils.closeQuietly(is);
+		}
+	}
```

Comment:
So this method is copied all over the place, but I note that we already have [a standard implementation of the exact code](https://github.com/mekomsolutions/openmrs-module-commonreports/blob/71ab813f5de1e7c23480106d2f02a1bc0161740f/api/src/main/java/org/openmrs/module/commonreports/common/Helper.java#L13). It seems like it would be better to remove all these extraneous `getSqlString()` method definitions and just use the `getStringFromResource()` method.